src/payment/strategies/amazon-pay/amazon-pay-payment-strategy.ts

```diff
@@ -92,6 +94,10 @@ export default class AmazonPayPaymentStrategy implements PaymentStrategy {
             throw new InvalidArgumentError('Unable to proceed because "payload.payment.methodId" argument is not provided.');
         }
 
+        if (!this._isPaymentMethodSelected) {
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
```

Comment:
Are you sure this is the right error to throw? As far I know, `_isPaymentMethodSelected` would be false if the shopper hasn't selected a credit card option, or we are not able to save the billing address provided by Amazon, not purely because the developer using this library has forgotten to `initialize` the payment method. You can either create a new custom error type that's more specific or use an existing one such as `PaymentMethodFailedError`.

Reply:
Hi @davidchin thanks for the observation, I did not found the `PaymentMethodFailedError` but one that I think works perfectly for this case which is: `MissingPaymentMethod`